core-web/libs/portlets/edit-ema/portlet/src/lib/store/features/editor/toolbar/withEditorToolbar.ts

```diff
@@ -0,0 +1,199 @@
+ *
+ * @export
+ * @return {*}
+ */
+export function withEditorToolbar() {
+    return signalStoreFeature(
+        {
+            state: type<UVEState>()
+        },
+        withState<EditorToolbarState>(initialState),
+        withComputed((store) => ({
+            $toolbarProps: computed<ToolbarProps>(() => {
+                const params = store.params();
+                const url = sanitizeURL(params.url);
+
+                const pageAPIQueryParams = createPageApiUrlWithQueryParams(url, params);
+                const pageAPIResponse = store.pageAPIResponse();
+                const experiment = store.experiment?.();
+
+                const pageAPI = `/api/v1/page/${
+                    store.isTraditionalPage() ? 'render' : 'json'
+                }/${pageAPIQueryParams}`;
+
+                const isExperimentRunning = experiment?.status === DotExperimentStatus.RUNNING;
+                const shouldShowUnlock =
+                    pageAPIResponse.page.locked && pageAPIResponse.page.canLock;
+
+                const unlockButton = {
+                    inode: pageAPIResponse.page.inode,
+                    loading: store.status() === UVE_STATUS.LOADING
+                };
+                const shoudlShowInfoDisplay =
+                    !store.canEditPage() ||
+                    pageAPIResponse.page.locked ||
+                    !!store.device() ||
+                    !!store.socialMedia();
+
+                const bookmarksUrl = createFavoritePagesURL({
+                    languageId: Number(params.language_id),
+                    pageURI: url,
+                    siteId: pageAPIResponse.site.identifier
+                });
+
+                return {
+                    bookmarksUrl,
+                    copyUrl: createPureURL(params),
+                    apiUrl: `${window.location.origin}${pageAPI}`,
+                    currentLanguage: pageAPIResponse.viewAs.language,
+                    urlContentMap: store.isEditState()
+                        ? (pageAPIResponse.urlContentMap ?? null)
+                        : null,
+                    runningExperiment: isExperimentRunning ? experiment : null,
+                    workflowActionsInode: store.canEditPage() ? pageAPIResponse.page.inode : null,
+                    unlockButton: shouldShowUnlock ? unlockButton : null,
+                    showInfoDisplay: shoudlShowInfoDisplay,
+                    deviceSelector: {
+                        apiLink: `${params.clientHost ?? window.location.origin}${pageAPI}`,
+                        hideSocialMedia: !store.isTraditionalPage()
+                    },
+                    personaSelector: {
+                        pageId: pageAPIResponse.page.identifier,
+                        value: pageAPIResponse.viewAs.persona ?? DEFAULT_PERSONA
+                    }
+                };
+            }),
+            $infoDisplayOptions: computed<InfoOptions>(() => {
```

Comment:
I think we can optimize this.

Not sure how to do it right now but we should keep an eye on this